File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    const set = new Set('abcdefghijklmnopqrstuvwxyz');
    skip.split('').forEach((char) => set.delete(char));
    const arr = [...set];
    return s.split('').map((char) => {
        const newIdx = arr.indexOf(char) + index;
        console.log(newIdx)
        return arr[newIdx % set.size];
    }).join('');
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ''    
    alphabet = 'abcdefghijklmnopqrstuvwxyz'
    for sk in skip:
        alphabet= alphabet.replace(sk, '')

    for i in range(len(s)):
        originalIndex = alphabet.index(s[i])
        newIndex = (originalIndex + index) % len(alphabet)
        answer += alphabet[newIndex]
        
    return answer